plugins/aggregators/basicstats/basicstats.go

```diff
@@ -152,6 +156,10 @@ func (b *BasicStats) Add(in telegraf.Metric) {
 				}
 				// last compute
 				tmp.last = fv
+				// first compute
+				if tmp.count == 1 {
+					tmp.first = fv
+				}
```

Comment:
Looking closer to the code I think you can skip this completely! The value is set above if the metric is not yet in the cache and after that the count will increase so we never choose the code-path... Could you confirm this can be removed and remove it? Sorry for not catching it in the first pass...

Reply:
You are right. New change